lib/resources/oracledb_session.rb

```diff
@@ -68,11 +68,11 @@ def query(q)
       query = verify_query(escaped_query)
       query += ';' unless query.end_with?(';')
       if @db_role.nil?
-        command = %{#{bin} "#{@user}"/"#{@password}"@#{@host}:#{@port}/#{@service} <<EOC\n#{opts}\n#{query}\nEXIT\nEOC}
+        command = %{#{bin} -S "#{@user}"/"#{@password}"@#{@host}:#{@port}/#{@service} <<EOC\n#{opts}\n#{query}\nEXIT\nEOC}
```

Comment:
A set of comments picking these apart - or links to online Oracle docs - would help the maintainability of this code. A lot happens in different ways, and I don't see the reasoning.

Reply:
This logic was added in the previous PR that touched this resource: https://github.com/inspec/inspec/pull/3170. The way the logic is constructed appears to do:
1. User & password authentication
2. User & password authentication plus assuming a database role
3. User & password authentication plus assuming a database role executing the binary as root

Reply:
Yes, and I found it confusing then, as well.  What I was asking for here is comments *in code* explaining what is going on, or a URL *in a code comment* where a maintainer with little knowledge of Oracle can go to understand what is happening, and why it happens three different ways.